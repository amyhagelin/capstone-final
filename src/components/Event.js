import React from "react";
import moment from "moment";


const Event = (props) => (
    <div className="single-event">
        { moment.utc(props.event.date).format("MMMM D, YYYY") } | { props.event.time } | { props.event.location } | { props.event.type }
        <br/>
        Triggers: { props.event.triggers } <br/>
        Medication: { props.event.medication } <br/>
        Notes: { props.event.notes }
    </div>
)

export default Event