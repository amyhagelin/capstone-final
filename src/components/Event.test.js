import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Event from './Event';

describe ('<Event />', () => {
    it('renders correctly', () => {
         const event = {
            date: new Date(123),
            time: '6:00 pm',
            type: 'aura',
            triggers: 'xyz',
            location: 'home',
            medication: 'abc',
            notes: 'def'
        };
        const tree = renderer.create(
            <Event event={ event } />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // it('Renders without crashing', () => {
    //     shallow(<Event />);
    // });

    // it('Renders with proper data', () => {
    //     const event = {
    //         date: new Date(123),
    //         time: '5:00 pm',
    //         type: 'aura',
    //         triggers: 'xyz',
    //         location: 'home',
    //         medication: 'abc',
    //         notes: 'def'
    //     };

    //     const wrapper = shallow(<Event event={ event } />);

    //     expect(wrapper.find('h2').length).toEqual(1)
    // });


});