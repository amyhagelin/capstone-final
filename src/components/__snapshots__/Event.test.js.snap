// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> renders correctly 1`] = `
<div
  className="single-event"
>
  January 1, 1970
   | 
  5:00 pm
   | 
  home
   | 
  aura
  <br />
  Triggers: 
  xyz
   
  <br />
  Medication: 
  abc
   
  <br />
  Notes: 
  def
</div>
`;
